packages/commonwealth/client/scripts/state/api/threads/fetchThreads.ts

```diff
@@ -0,0 +1,226 @@
+import { useInfiniteQuery, useQuery } from '@tanstack/react-query';
+import axios from 'axios';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import app from 'state';
+import { ApiEndpoints } from 'state/api/config';
+import {
+  ThreadTimelineFilterTypes
+} from '../../../models/types';
+import Thread from 'models/Thread';
+import { cacheTypes } from './helpers/cache';
+
+const THREADS_STALE_TIME = 100 * 1_000; // 30s todo: UNDO THIS
```

Comment:
What should be the optimal time for cache to be valid? I would suggest 3 mins but curious to what others think @jnaviask / @CowMuon / @masvelio

Reply:
@mzparacha what is the impact of a stale cache in this case? can you explain a little more deeply?

Reply:
nice readings

https://tkdodo.eu/blog/react-query-as-a-state-manager#customize-staletime and https://tkdodo.eu/blog/practical-react-query#the-defaults-explained

tldr

> As long as data is fresh, it will always come from the cache only. You will not see a network request for fresh data, no matter how often you want to retrieve it.

In my opinion 3 min is too long. it means that we won't see new threads sooner than every 3 minutes. Something around 1 min sounds good to me, maybe even less

Reply:
Will update here soon

Reply:
> @mzparacha what is the impact of a stale cache in this case? can you explain a little more deeply?

Firstly I made sure to remain as close to the existing code as possible to avoid any unnecessary issues.

The existing code allowed for an indefinite cache i.e. the data is there unless you refresh. Now we are refreshing it after every 3 mins.

Reply:
> nice readings
> 
> https://tkdodo.eu/blog/react-query-as-a-state-manager#customize-staletime and https://tkdodo.eu/blog/practical-react-query#the-defaults-explained
> 
> tldr
> 
> > As long as data is fresh, it will always come from the cache only. You will not see a network request for fresh data, no matter how often you want to retrieve it.
> 
> In my opinion 3 min is too long. it means that we won't see new threads sooner than every 3 minutes. Something around 1 min sounds good to me, maybe even less

Hmm, there is a slight distinction here.

> As long as data is fresh, it will always come from the cache only. You will not see a network request for fresh data, no matter how often you want to retrieve it.

this in our case means after every 3 mins react query will make a new API call, but in those 3 mins no matter how many times we call RQ API it will not trigger network request, unless we invalidate cache (in which case, it will make a new API call)

Reply:
I'm still worried about this cache length. Consider the following:
1. Alice loads the threads page.
2. Bob creates a thread.
3. Alice refreshes -- where is the thread?

We need to make sure that when someone makes a post, it appears for others basically immediately. Not sure what we need to do to support this, but I don't like the idea of having to wait 3 minutes to see e.g. your posts in my forum. @mzparacha

Reply:
cc @mzparacha